R/plot_step_ahead_forecasts.R

```diff
@@ -0,0 +1,798 @@
+            line_color = ens_color, opacity = fill_transparency,
+            top_layer = top_layer, show_truth_legend = FALSE,
+            interactive = TRUE)
+        } else {
+          plot_model <- simple_model_plot(
+            plot_model, df_point_ens, df_ribbon_ens, FALSE, truth_data,
+            line_color = ens_color, opacity = fill_transparency,
+            showlegend = FALSE, top_layer = top_layer,
+            show_truth_legend = FALSE, interactive = TRUE)
+        }
+      }
+      if (!is.null(facet_title)) {
+        if (grepl("top", facet_title)) {
+          y_title <- 1
+          y_anchor <- "top"
+        } else if  (grepl("bottom", facet_title)) {
+          y_title <- 0
+          y_anchor <- "bottom"
+        }
+        if (grepl("left", facet_title)) {
+          x_title <- 0
+          x_anchor <- "left"
+        } else if  (grepl("right", facet_title)) {
+          x_title <- 1
+          x_anchor <- "right"
+        }
+        plot_model <- plotly::layout(
+          plot_model,
+          annotations = list(x = x_title, y = y_title, xref = "paper",
+                             yref = "paper", xanchor = x_anchor,
+                             yanchor = y_anchor, showarrow = FALSE, text = x))
+      }
+      return(plot_model)
+    })
+    plot_model <- plotly::subplot(subplot, nrows = facet_nrow, shareX = sharex,
+                                  shareY = sharey)
+  } else {
+    df_point <- all_plot$median
+    df_ribbon <- all_plot[names(all_plot) %in% intervals]
+    if (!is.null(all_ens)) {
+      df_point_ens <- all_ens$median
+      df_ribbon_ens <- all_ens[names(all_ens) %in% intervals]
+    }
+    plot_model <- simple_model_plot(plot_model, df_point, df_ribbon, plot_truth,
+                                    truth_data, opacity = fill_transparency,
+                                    top_layer = top_layer,
+                                    interactive = interactive)
+
+    # Ensemble color
+    if (!is.null(all_ens)) {
+      plot_model <- simple_model_plot(
+        plot_model, df_point_ens, df_ribbon_ens, truth_data = truth_data,
+        plot_truth = FALSE, opacity = fill_transparency, line_color = ens_color,
+        top_layer = top_layer, interactive = interactive)
+    }
+    if (!is.null(facet)) {
+      plot_model <-  plot_model +
+        ggplot2::facet_wrap(facet, nrow = facet_nrow, ncol = facet_ncol,
+                            scales = facet_scales)
+    }
+  }
+
+  if (!interactive) {
+    plot_model <- plot_model +
+      scale_color_manual(values = pal_value, name = "Legend") +
+      scale_fill_manual(values = pal_value, name = "Legend")
+  }
+  return(plot_model)
+}
+
+
+#' Basic Plot for model outputs
+#'
+#' Create a simple Plotly time-series plot for model projection outputs.
+#'
+#'@param forecast_data a `model_output_df` object, containing all the required
+#' columns, and a "target_date" and a "model_id" column.
```

Comment:
Why not leave this more flexible, and allow someone to specify a column name that points to the variable that should be mapped to the x-axis?

Reply:
That's a good idea, I can add a `x_col_name` parameter with a default to "target_date".

Reply:
As I am implementing it, I wonder if it makes sense to have the same parameter for the truth_date, with for example a `x_truth_col_name` parameter set by default on "time_idx"